homeassistant/components/generic_thermostat/climate.py

```diff
@@ -342,9 +343,12 @@ async def async_set_hvac_mode(self, hvac_mode):
     async def async_set_temperature(self, **kwargs):
         """Set new target temperature."""
         temperature = kwargs.get(ATTR_TEMPERATURE)
+        hvac_mode = kwargs.get(ATTR_HVAC_MODE)
         if temperature is None:
             return
         self._target_temp = temperature
+        if hvac_mode is not None:
+            await self.async_set_hvac_mode(hvac_mode=hvac_mode)
```

Comment:
How will the multiple subsequent calls to `_async_control_heating` affect the integration?

Reply:
Not sure what potential problems this might cause. Many users on the forum reported this works as expected, for example here: https://community.home-assistant.io/t/scheduler-card-custom-component/217458/927?u=misiu
I'll add a test for both temperature and HVAC mode.
The same method is used in intesishome and fritzbox.

Reply:
We need to think about the consequences from this change. Someone needs to present an analysis around this before we can merge this.

Reply:
I'll ask on the forum. Some users applied this as a custom component.

Reply:
@MartinHjelmare did you look at the changes @Misiu made to the component?
Only thing he added is a check if a `hvac_mode` parameter is provided in the `set_temperature` service call, and call the `set_hvac_mode` function if so. This change is very harmless, fully backwards compatible, and tested _by use_.

It also makes the implementation compatible with the [HA definition](https://www.home-assistant.io/integrations/climate/#service-climateset_temperature)  for the set_temperature service.
The effect is literally the same as sequentially calling the `set_hvac_mode` and `set_temperature` services.

I noticed that a lot of climate integrations (3 out of 4) actually lack support for setting `hvac_mode` together with `set_temperature`. I recommend to plan a big sweep to bring all these components up to par.
It's quite annoying since the support for this feature is not detectable through the `supported_features` (the attribute would simply be ignored).

Anyway, I think its perfectly safe to merge this change. I cannot support this claim through analysis though...